multiqc/modules/bamdst/bamdst.py

```diff
@@ -0,0 +1,419 @@
+            },
+            "[Total] Fraction of Mapped Reads": {
+                "title": "Mapped",
+                "description": f"Fraction of mapped reads in all reads ({config.read_count_desc})",
+                "scale": "PuBu",
+                "min": 0.0,
+                "max": 100.0,
+                "suffix": "%",
+                "modify": lambda x: x * 100.0,
+                "hidden": True,
+            },
+            "[Target] Target Reads": {
+                "title": f"Target reads ({config.read_count_prefix})",
+                "description": f"Reads mapped on target ({config.read_count_desc})",
+                "scale": "RdYlGn",
+                "min": 0,
+                "modify": lambda x: x * config.read_count_multiplier,
+                "hidden": True,
+            },
+            "[Target] Fraction of Target Reads in all reads": {
+                "title": "% in all",
+                "description": "Fraction of target reads in all reads",
+                "scale": "PuBu",
+                "min": 0.0,
+                "max": 100.0,
+                "suffix": "%",
+                "modify": lambda x: x * 100.0,
+                "hidden": True,
+            },
+            "[Target] Fraction of Target Reads in mapped reads": {
+                "title": "% in mapped",
+                "description": "Fraction of target reads in mapped reads",
+                "scale": "PuBu",
+                "min": 0.0,
+                "max": 100.0,
+                "suffix": "%",
+                "modify": lambda x: x * 100.0,
+                "hidden": True,
+            },
+            "[Target] Average depth": {
+                "title": "Avg depth",
+                "description": "Average depth, on target",
+                "min": 0,
+                "suffix": "X",
+                "scale": "BuPu",
+            },
+            "[Target] Len of region": {
+                "title": "Region len",
+                "description": "Length of target region",
+                "min": 0,
+                "format": "{:,d}",
+                "suffix": "&nbsp;bp",
+                "scale": "Greys",
+                "hidden": True,
+            },
+        }
+
+        coverage_metrics = []
+        for s_name, d in data_by_sample.items():
+            for k in d.keys():
+                if k.startswith("[Target] Coverage (") and k not in coverage_metrics:
+                    coverage_metrics.append(k)
+        for m in coverage_metrics:
+            short = m.replace("[Target] Coverage (", "").replace(")", "")
+            headers[m] = {
+                "title": short,
+                "description": f"Target coverage ({short})",
+                "min": 0,
+                "suffix": "%",
+                "format": "{:.2f}",
+                "scale": "RdYlGn",
+                "modify": lambda x: x * 100.0,
+                "hidden": short != ">0x",
+            }
+
+        self.general_stats_addcols(data_by_sample, headers)
```

Comment:
As with the other module, I don't love just sticking all columns into General Stats. If we do this a lot, or if there are many samples, it'll bloat the report size. It also makes the "Configure Columns" dialogue massive and difficult to use if there are hundreds of columns to choose from..